packages/components/src/utils/unit-values.ts

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { isNil } from 'lodash';
+
+const UNITED_VALUE_REGEX = /[\d.\-+]*\s*(fr|cm|mm|Q|in|pc|pt|px|em|ex|ch|rem|lh|vw|vh|vmin|vmax|%|cap|ic|rlh|vi|vb|deg|grad|rad|turn|s|ms|Hz|kHz|dpi|dpcm|dppx)$/;
```

Comment:
It looks like something like `10notaunit rem` would generate a positive match — maybe adding `^` at the very start of the Reg Exp would prevent this edge case from happening?

Reply:
I did that but it's too restrictive. I think we might want to just change the whole thing to be a regex match instead of parsing the number using parseFloat's weird implementation.

Reply:
I believe this is a good approach, a bit more restrictive but fair